### PR TITLE
Remove random runner from build tools (internal) tests

### PR DESCRIPTION
**Pull request by breskeby**
This removes unrequited overhead for writing build tool and build tool internal tests and mechanically ports all junit3 tests to junit4.

### CODE REVIEW

File: build-tools-internal/src/test/java/org/elasticsearch/gradle/internal/test/rest/transform/match/AddMatchTests.java [hunk 1]
```diff
@@ -34,7 +40,7 @@ public void testAddAllNotSupported() throws Exception {
         JsonNode addNode = MAPPER.convertValue("_doc", JsonNode.class);
         assertEquals(
             "adding matches is only supported for named tests",
-            expectThrows(
+            Assert.assertThrows(
```
Comment (pugnascotia) [inline build-tools-internal/src/test/java/org/elasticsearch/gradle/internal/test/rest/transform/match/AddMatchTests.java:43]: Should we statically import `assertThrows`, as with other methods on `Assert`?

**Review by pugnascotia — APPROVED** [review]
LGTM.

So do we not need things like `@ThreadLeakFilters` now?

**Comment by breskeby** [comment]
> ThreadLeakFilters



> LGTM.
> 
> So do we not need things like `@ThreadLeakFilters` now?

Newer versions of gradle should handle this properly. Also we have ported / wrote most integration tests now using Spock not using that filter for a while. I haven't seen any issues. Unit tests should not have ran into problems here anyhow.

**Review by mark-vieira — APPROVED** [review]
LGTM

**Pull request merged**